File: code_folder/monitoring/push_prometheus.go
package main

import (
  "fmt"
  "net/http"

  "github.com/prometheus/client_golang/prometheus"
  "github.com/prometheus/client_golang/prometheus/promauto"
  "github.com/prometheus/client_golang/prometheus/promhttp"
)

var (
  // Create a Histogram metric to track the request latencies
  requestLatency = promauto.NewHistogram(prometheus.HistogramOpts{
    Name: "request_latency",
    Help: "Request latencies",
  })
)

func main() {
  // Register the metrics with Prometheus
  prometheus.MustRegister(requestLatency)

  // Collect some data for the metric
  requestLatency.Observe(0.3)
  requestLatency.Observe(0.5)

  // Start the HTTP server to expose the metrics
  http.Handle("/metrics", promhttp.Handler())
  fmt.Println("Starting server on port 8080")
  http.ListenAndServe(":8080", nil)
}

// The above code will expose the metrics at http://localhost:8080/metrics. You can see the metrics in the browser or using curl:

File: code_folder/monitoring/push_prometheus.py
from urllib import response
from prometheus_client import Summary, push_to_gateway

# Create a Summary metric to track the average response time
response_time = Summary("response_time", "Average response time")


# Record the response time for a specific request
@response_time.time()
def handle_request():
    # Perform some work and return the response
    return response


# Push the collected metrics to the Prometheus gateway
push_to_gateway("prometheus-server:9090", job="my-app")
